### PR TITLE
Update Dataflow instructions to Java 8 and Cloud Shell.

### PR DESCRIPTION
**Pull request by deflaux**
Also fixes https://github.com/googlegenomics/readthedocs/issues/128, https://github.com/googlegenomics/readthedocs/issues/119, and https://github.com/googlegenomics/readthedocs/issues/115.

Staged on http://googlegenomics.readthedocs.io/en/staging

Note that this should not be merged until after https://github.com/googlegenomics/dataflow-java/pull/187 is released.

### CODE REVIEW

File: docs/source/includes/alpn_setup.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+If you want to run a small pipeline on your machine before running it in parallel on Compute Engine, you will need `ALPN`_ since many of these pipelines require it.  When running locally, this must be provided on the boot classpath but when running on Compute Engine Dataflow workers this is already configured for you. You can download the correct version from `here <http://mvnrepository.com/artifact/org.mortbay.jetty.alpn/alpn-boot>`__.  For example::
```
Comment (mbookman) [inline docs/source/includes/alpn_setup.rst]: I am unclear about the phrase "correct version". Is there something specific implied there that needs elaboration?
Or should this just say "You can download ALPN from `here ...`"?

Comment (deflaux) [inline docs/source/includes/alpn_setup.rst]: Agreed, will fix.  That's left over language from when this talked about both Java 7 and 8.

File: docs/source/includes/dataflow_on_gce_setup.rst [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
 
   sudo apt-get update
-  sudo apt-get install --assume-yes openjdk-7-jdk maven
+  sudo apt-get install --assume-yes openjdk-8-jdk maven
   sudo update-alternatives --config java
+  sudo update-alternatives --config javac
+
+.. note::
+
+  Depending on the pipeline, Cloud Shell may not not have sufficient memory to run pipeline locally (e.g., without the ``--runner`` command line flag).  If you get error ``java.lang.OutOfMemoryError: Java heap space``, follow the instructions to run the pipeline on Compute Engine Dataflow workers instead of locally (e.g. use ``--runner=DataflowPipelineRunner``).
```
Comment (mbookman) [inline docs/source/includes/dataflow_on_gce_setup.rst]: s/to run pipeline/to run the pipeline

Comment (mbookman) [inline docs/source/includes/dataflow_on_gce_setup.rst]: "on Compute Engine Dataflow workers" sounds odd to me.

"on Compute Engine" or "using Compute Engine Dataflow workers" sounds better.

**Comment by deflaux** [comment]
updated!  PTAL

**Comment by mbookman** [comment]
LGTM

**Pull request merged**